Apps/WebClient/src/ClientApp/src/store/modules/notification/actions.ts

```diff
@@ -0,0 +1,130 @@
+import { ErrorSourceType, ErrorType } from "@/constants/errorType";
+import ApiResult from "@/models/apiResult";
+import { ResultError } from "@/models/errors";
+import Notification from "@/models/notification";
+import { LoadStatus } from "@/models/storeOperations";
+import container from "@/plugins/container";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import { ILogger, INotificationService } from "@/services/interfaces";
+
+import { NotificationActions } from "./types";
+
+export const actions: NotificationActions = {
+    retrieve(
+        context,
+        params: { hdid: string }
+    ): Promise<ApiResult<Notification[]>> {
+        const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+        const notificationService = container.get<INotificationService>(
+            SERVICE_IDENTIFIER.NotificationService
+        );
+
+        return new Promise((resolve, reject) => {
+            const notifications: Notification[] = context.getters.notification;
+            if (context.state.status === LoadStatus.LOADED) {
+                logger.debug(`Notifications found stored, not querying!`);
+                resolve({
+                    resourcePayload: notifications,
+                });
+            } else {
+                logger.debug(`Retrieving Notifications`);
+                context.commit("setRequested");
+                notificationService
+                    .getNotifications(params.hdid)
+                    .then((result) => {
+                        if (result.resourcePayload) {
+                            context.commit(
+                                "setNotifications",
+                                result.resourcePayload
+                            );
+                            resolve(result);
+                        }
+                    })
+                    .catch((error: ResultError) => {
+                        context.dispatch("handleError", {
+                            error,
+                            errorType: ErrorType.Retrieve,
+                        });
+                        reject(error);
+                    });
+            }
+        });
+    },
+    deleteNotification(
```

Comment:
Corresponding backend code has been named dismiss.  Was there a reason to not keep things consistent.